test/integration/cni/pod_egress_traffic_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cni
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	k8sUtils "github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+)
+
+const (
+	cmdForPodEgressIpv6 = `ip -f inet6 address show dev v6if0 | sed -n "s/.*inet6 \(fd00::ac:.*\)\/118.*/\1/p"`
+	cmdForPodEgressIpv4 = `ip -f inet address show dev v4if0 | sed -n "s/.*inet \(169.254..*\)\/22.*/\1/p"`
+)
+
+// Verifies connectivity to deployment behind different service types
+var _ = Describe("[CANARY] test cluster egress connectivity", func() {
+	var (
```

Comment:
nit: these feel like they should be defined in the global space

Reply:
You mean the vars defined inside `Describe`, putting them into global means we encourage test specs to share those variables. This introduce several questions: 

Who initialize them (or clean them up):
1.  if we initialize them in suite, but different spec has different requirements to them. For example, different specs need different tester container and different deployments.
2  if we initialize them in test spec, there is no benefit for them to be global, and global vars tangle those test specs together

One benefit I can see by putting them into global is if we can re-use/shared by different test specs, but currently, each test spec has special settings.

There is only one var `isIPv4Cluster` is a candidate for global var, but currently only our test spec is using it.